internal/jobtracker/jobtracker_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2025 Canonical.
+package jobtracker_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	qt "github.com/frankban/quicktest"
+	"github.com/frankban/quicktest/qtsuite"
+
+	"github.com/canonical/jimm/v3/internal/db"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/jobtracker"
+	"github.com/canonical/jimm/v3/internal/testutils/jimmtest"
+)
+
+type jobTrackerSuite struct {
+	db      *db.Database
+	tracker *jobtracker.Tracker
+}
+
+func (s *jobTrackerSuite) Init(c *qt.C) {
+	db := &db.Database{
+		DB: jimmtest.PostgresDB(c, time.Now),
+	}
+	err := db.Migrate(context.Background())
+	c.Assert(err, qt.IsNil)
+
+	s.db = db
+	tracker, err := jobtracker.NewJobTracker(db, time.Second*5)
+	c.Assert(err, qt.IsNil)
+	s.tracker = tracker
+}
+
+// !!! --- For PR reviewers, currently we're polling the job status as we are waiting
+// for the forselect to update. We could use gomock / some mock and capture the call. What would you prefer?
+// Gomock would allow us to see the order of calls and expected params, so perhaps a bit cleaner.
+//
+// Or we leave it polling.
+func (s *jobTrackerSuite) TestJobTracker_Run_JobIsSetFailed_JobError(c *qt.C) {
+	testCtx := c.Context()
+
+	aFastDyingJob := func(ctx context.Context) error {
+		return errors.New("I died really fast")
+	}
+
+	id, err := s.tracker.Run(testCtx, "test-job-type", aFastDyingJob, time.Second*1000)
+	c.Assert(err, qt.IsNil)
+
+	var status dbmodel.JobStatus
+	var pollerr error
```

Comment:
What do you think about testing `j.manageJob()` which will return only when the job completes.
If you only want to test the exported method, then I think the polling is fine since you can't know when the job is done.

Reply:
Polling for the job to complete could be moved to a helper method as I see its reused.

Reply:
Agreed, I'll make helper.

And we could with a mocked store. I think I'd remove the testing of Run then though? But at that point, I'd probably just test Run with a mock?